test/functional/normal/put_spec.lua

```diff
@@ -24,16 +24,17 @@ local function reset()
   funcs.setreg('a', 'test_stringa', 'V')
   funcs.setreg('b', 'test_stringb\ntest_stringb\ntest_stringb', 'b')
   funcs.setreg('"', 'test_string"', 'v')
+  execute('set virtualedit=')
 end
 
 -- We check the last inserted register ". in each of these tests because it is
 -- implemented completely differently in do_put().
 -- It is implemented differently so that control characters and imap'ped
 -- characters work in the same manner when pasted as when inserted.
 describe('put command', function()
-  -- Put a call to clear() here to force the connection to the server.
-  -- This means we can use the funcs.*() functions while mangling text before
-  -- the actual tests are run.
+  -- This call to clear() here not only ensures a clean state and forces the
+  -- connection to the server. This means we can use the funcs.*() functions
+  -- while mangling text before the actual tests are run.
```

Comment:
This comment is confusing, would be less confusing to have nothing at all. If anyone tries to remove the clear() for some reason they'll quickly learn the reason :)